### PR TITLE
Finetuning & model saving script

### PR DESCRIPTION
**Pull request by gabrieltseng**
- [x] Finetune a TIML model for any country and target label, and save it
- [x] Include the normalizing dictionary when saving the model

cc: https://github.com/nasaharvest/timl/pull/6#issuecomment-1067171438

### CODE REVIEW

File: crop_classification/finetune_timl.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    uses_encoder = (data_folder / DL_TIML / "encoder_state_dict.pth").exists()
+
+    model = load_timl_model(
+        task_info=task_info,
+        input_size=dataset.num_bands
+        if uses_encoder
+        else dataset.num_bands + task_info.shape[0],
+        num_timesteps=num_timesteps,
+        model_state_dict_path=data_folder / DL_TIML / "state_dict.pth",
+        encoder_state_dict_path=data_folder / DL_TIML / "encoder_state_dict.pth",
```
Comment (ivanzvonkov) [inline crop_classification/finetune_timl.py:95]: What do you think about passing the `normalizing_dict` in here?
(And then doing the numpy -> list conversion inside)
I think it goes along with passing passing the bands and timesteps from the dataset

**Comment by ivanzvonkov** [comment]
Am I correctly understanding that the model is being fine-tuned on just a training dataset here? Should we consider using a portion of the available tasks data for validation?

File: crop_classification/finetune_timl.py (hunk 1 above)
Comment (gabrieltseng) [reply crop_classification/finetune_timl.py:95]: yup - this makes sense. Updated

**Review by ivanzvonkov — APPROVED** [review]

**Comment by ivanzvonkov** [comment]
Any reason to not include a trained model in this PR?

**Comment by gabrieltseng** [comment]
> Any reason to not include a trained model in this PR?

Let's discuss in person exactly how we can save models - I'd rather not add them directly to git, since I can see this getting cluttered quickly (and also we'd need / want some way to pin the model to the cropharvest version it was trained against)
